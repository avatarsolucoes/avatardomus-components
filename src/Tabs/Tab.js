/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import css from './tabs.css';

export default function Tab(props) {
  const { className, label, isActive, onClick, onClickTab } = props;

  const cfgClassLink = {
    [css.tabsTabLink]: true,
    [css.active]: isActive,
  };

  const activeLinkClass = cx(cfgClassLink, 'tabs__tab-link', className, {
    active: isActive,
  });

  const tabClass = cx(css.tabsTab, 'tabs__tab', className);

  function handleClick(e) {
    onClickTab(e);
    onClick(e);
  }

  return (
    <div className={tabClass}>
      <button type="button" className={activeLinkClass} onClick={handleClick}>
        {label}
      </button>
    </div>
  );
}

Tab.propTypes = {
  label: PropTypes.string,
  className: PropTypes.string,
  isActive: PropTypes.bool,
  onClick: PropTypes.func,
  onClickTab: PropTypes.func,
};

Tab.defaultProps = {
  label: `\u0020`,
  className: null,
  isActive: false,
  onClick: () => {},
  onClickTab: null,
};
