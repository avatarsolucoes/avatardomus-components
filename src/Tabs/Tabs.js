/* eslint-disable react/forbid-prop-types */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import css from './tabs.css';

export default function Tabs(props) {
  const { className, defaultActiveIndex } = props;
  const [activeIndex, setActiveIndex] = useState(defaultActiveIndex);

  function handleTabClick(tabIndex) {
    if (tabIndex !== activeIndex) {
      setActiveIndex(tabIndex);
    }
  }

  function cloneTabElement(tab, index = 0) {
    return React.cloneElement(tab, {
      onClick: () => {
        handleTabClick(index);
      },
      tabIndex: index,
      isActive: index === activeIndex,
    });
  }

  function renderChildrenTabs() {
    const { children } = props;
    if (!Array.isArray(children)) {
      return cloneTabElement(children);
    }
    return children.map(cloneTabElement);
  }

  function renderActiveTabContent() {
    const { children } = props;
    if (children[activeIndex]) {
      return children[activeIndex].props.children;
    }
    return children.props.children;
  }

  const cssClass = cx(css.tabs, 'tabs', className);
  const classUl = cx(css.tabsList, 'tabs__list');
  const classContent = cx(css.tabsContent, 'tabs__content');

  return (
    <section className={cssClass}>
      <ul className={classUl}>{renderChildrenTabs()}</ul>
      <div className={classContent}>{renderActiveTabContent()}</div>
    </section>
  );
}

Tabs.propTypes = {
  children: PropTypes.any.isRequired,
  className: PropTypes.string,
  defaultActiveIndex: PropTypes.number,
};

Tabs.defaultProps = {
  className: null,
  defaultActiveIndex: 0,
};
