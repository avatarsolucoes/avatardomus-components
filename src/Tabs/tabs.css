/* .ctrl-tab {
  position: relative;
  margin: 0 auto;
  padding: 0;
}

.ctrl-tab .ctrl-tab-header {
  display: flex;
  flex-flow: row nowrap;
  margin: 0;
}

.ctrl-tab .ctrl-tab-header label {
  cursor: pointer;
  display: block;
  padding: 4px;
  margin: 0;
} */

.tabs {
  width: 100%;
  max-width: 100%;
  display: block;
  margin: 0 auto;
  position: relative;
}

.tabs__list,
.tabsList {
  width: 100%;
  max-width: 100%;
  margin: 0 auto;
  border-bottom: 1px solid #FFF;
  padding: 0;
  display: flex;
  align-content: center;
  align-items: center;
  flex-flow: nowrap row;
}

.tabs__tab,
.tabsTab {
  display: block;
  max-width: 100%;
  padding: 0;
  margin: 0;
}

.tabs__tab-link,
.tabsTabLink {
  display: inline-block;
  text-align: center;
  background: transparent;
  border: 0;
  border-bottom: 0px solid transparent;
  cursor: pointer;
  text-decoration: none !important;
  font-size: 16px;
  color: #FFF;
  padding: 4px 8px;
  transition: all ease 0.2s;
}

.tabs__tab-link:hover,
.tabsTabLink:hover {
  background-color: transparent;
  border-bottom: 4px solid rgba(255, 255, 255, 0.2);
}

.tabs__tab-link.active,
.tabsTabLink.active {
  /*border-color:;*/
  border-bottom: 4px solid #FFF;
}

.tabs__content,
.tabsContent {
  margin: 4px auto;
  width: 100%;
}
