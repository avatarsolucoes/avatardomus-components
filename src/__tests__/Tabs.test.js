import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import { Tab, Tabs } from '../Tabs';

function RenderTabs() {
  return (
    <Tabs defaultActiveIndex={1}>
      <Tab label="A" key={`tab${1}`} onClick={() => {}}>
        A
      </Tab>
      <Tab label="B" key={`tab${2}`}>
        B
      </Tab>
      <Tab key={`tab${3}`}>C</Tab>
    </Tabs>
  );
}

function RenderTab() {
  return (
    <Tabs>
      <Tab
        label="A1"
        key={`tab${1}`}
        onClickTab={() => {
          console.log('test');
        }}
      >
        A1
      </Tab>
    </Tabs>
  );
}

describe('test Tabs', () => {
  it(`is truthy Span`, () => {
    expect(Tabs).toBeTruthy();
    expect(Tab).toBeTruthy();
  });

  it(`Test Span`, () => {
    const wrapper = {
      text: renderer.create(<Tabs />).toJSON(),
    };
    expect(wrapper).toMatchSnapshot();
  });

  it('should show App click', () => {
    const wrapper = mount(<RenderTabs />);
    wrapper
      .find('.tabs__tab-link')
      .at(1)
      .simulate('click');
    wrapper.update();
    wrapper
      .find('.tabs__tab-link')
      .at(0)
      .simulate('click');
    wrapper.update();
    wrapper
      .find('.tabs__tab-link')
      .at(2)
      .simulate('click');
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it(`Test shallow `, () => {
    const wrapper = shallow(<RenderTab />).html();
    expect(wrapper).toMatchSnapshot();
  });
});
