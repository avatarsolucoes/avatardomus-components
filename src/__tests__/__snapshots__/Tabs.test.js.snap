// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Tabs Test Span 1`] = `
Object {
  "text": <section
    className="tabs"
  >
    <ul
      className="tabs__list"
    />
    <div
      className="tabs__content"
    />
  </section>,
}
`;

exports[`test Tabs Test shallow  1`] = `"<section class=\\"tabs\\"><ul class=\\"tabs__list\\"><div class=\\"tabs__tab\\"><button type=\\"button\\" class=\\"undefined tabs__tab-link active\\">A1</button></div></ul><div class=\\"tabs__content\\">A1</div></section>"`;

exports[`test Tabs should show App click 1`] = `ReactWrapper {}`;
